src/main/java/com/puppycrawl/tools/checkstyle/site/XdocsTemplateSink.java

```diff
@@ -49,6 +77,51 @@ public class XdocsTemplateSink extends XdocSink {
     public XdocsTemplateSink(Writer writer, String encoding) {
         super(writer);
         this.encoding = encoding;
+        this.writer = new PrintWriter(writer);
+        tableContentWriterStack = new LinkedList<>();
```

Comment:
Creation of such stacks will be problematic.
As they are completely handled by child class. If we have no way to get access to abstract class fields, we should not do  fake sticks in our class, it it just better to claim in javadoc that we skipped all stacks as we do not need them for now and we do not have access to them in override.

But if one day something goes wrong, this override might be a problem, so maintainers need to be aware of it

Reply:
Done.